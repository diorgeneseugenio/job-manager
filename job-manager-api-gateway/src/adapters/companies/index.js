import got from "got";

const COMPANIES_API_URI = "http://job-manager-api:8800";

export default class CompaniesApi {
  static async fetchAllCompanies() {
    const body = await got.get(`${COMPANIES_API_URI}/companies`).json();
    return body;
  }

  static async createCompany({ logo, name, aboutDescription, city, state }) {
    const body = await got
      .post(`${COMPANIES_API_URI}/companies`, {
        json: { logo, name, aboutDescription, city, state },
      })
      .json();
    return body;
  }
}
