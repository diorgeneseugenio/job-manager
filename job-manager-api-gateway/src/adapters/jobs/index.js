import got from "got";

const JOBS_API_URI = "http://job-manager-api:8800";

export default class JobsApi {
  static async fetchAllJobs() {
    const body = await got.get(`${JOBS_API_URI}/jobs`).json();
    return body;
  }

  static async fetchJob(id) {
    const body = await got.get(`${JOBS_API_URI}/jobs/${id}`).json();
    return body;
  }

  static async createJob({
    companyId,
    title,
    description,
    salary,
    benefits,
    skills,
    allowRemote,
  }) {
    const body = await got
      .post(`${JOBS_API_URI}/jobs`, {
        json: {
          companyId,
          title,
          description,
          salary,
          benefits,
          skills,
          allowRemote,
        },
      })
      .json();
    return body;
  }

  static async updateJob(
    id,
    {
      companyId,
      title,
      description,
      salary,
      benefits,
      skills,
      allowRemote,
      ended,
    }
  ) {
    try {
      const body = await got
        .put(`${JOBS_API_URI}/jobs/${id}`, {
          json: {
            companyId,
            title,
            description,
            salary,
            benefits,
            skills,
            allowRemote,
            ended,
          },
        })
        .json();
      return body;
    } catch (e) {
      throw new Error(`Erro ao atualizar a vaga ${e}`);
    }
  }

  static async deleteJob(id) {
    try {
      const body = await got.delete(`${JOBS_API_URI}/jobs/${id}`).json();
      return body;
    } catch (e) {
      throw new Error(`Erro ao deletar a vaga ${e}`);
    }
  }

  static async filterByCompany(companyId) {
    const body = await got
      .get(`${JOBS_API_URI}/jobs/filterByCompany/${companyId}`)
      .json();
    return body;
  }
}
