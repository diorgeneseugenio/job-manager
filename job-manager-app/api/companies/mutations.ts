import { gql } from "@apollo/client";

export const CREATE_COMPANY = gql`
  mutation CreateCompany(
    $name: String!
    $aboutDescription: String!
    $state: String!
    $city: String!
  ) {
    createCompany(
      name: $name
      aboutDescription: $aboutDescription
      state: $state
      city: $city
    ) {
      id
      name
    }
  }
`;

export const UPDATE_COMPANY = gql`
  mutation UpdateCompany(
    $id: ID!
    $name: String
    $aboutDescription: String
    $state: String
    $city: String
  ) {
    updateCompany(
      id: $id
      name: $name
      aboutDescription: $aboutDescription
      state: $state
      city: $city
    )
  }
`;
