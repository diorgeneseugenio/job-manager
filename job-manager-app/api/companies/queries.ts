import { gql } from "@apollo/client";

export const FETCH_COMPANIES = gql`
  {
    fetchAllCompanies {
      id
      name
      aboutDescription
      state
      city
    }
  }
`;
