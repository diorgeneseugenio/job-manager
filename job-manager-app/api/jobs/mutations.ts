import { gql } from "@apollo/client";

export const CREATE_JOB = gql`
  mutation CreateJob(
    $companyId: ID!
    $title: String!
    $description: String!
    $salary: String!
    $benefits: String
    $skills: String
    $allowRemote: Boolean!
  ) {
    createJob(
      companyId: $companyId
      title: $title
      description: $description
      salary: $salary
      benefits: $benefits
      skills: $skills
      allowRemote: $allowRemote
    ) {
      id
      title
    }
  }
`;

export const UPDATE_JOB = gql`
  mutation UpdateJob(
    $id: ID!
    $companyId: ID
    $title: String
    $description: String
    $salary: String
    $benefits: String
    $skills: String
    $allowRemote: Boolean
    $ended: Boolean
  ) {
    updateJob(
      id: $id
      companyId: $companyId
      title: $title
      description: $description
      salary: $salary
      benefits: $benefits
      skills: $skills
      allowRemote: $allowRemote
      ended: $ended
    )
  }
`;

export const DELETE_JOB = gql`
  mutation DeleteJob($id: ID!) {
    deleteJob(id: $id)
  }
`;
