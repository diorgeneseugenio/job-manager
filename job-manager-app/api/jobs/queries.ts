import { gql } from "@apollo/client";

export const FETCH_JOBS = gql`
  {
    fetchAllJobs {
      id
      companyId
      title
      description
      salary
      benefits
      skills
      allowRemote
      ended
      company {
        name
        state
        city
      }
      createdAt
      updatedAt
    }
  }
`;

export const FILTER_BY_COMPANY = gql`
  query FilterByCompany($companyId: ID!) {
    filterByCompany(companyId: $companyId) {
      id
      companyId
      title
      description
      salary
      benefits
      skills
      allowRemote
      ended
      company {
        name
        state
        city
      }
      createdAt
      updatedAt
    }
  }
`;
