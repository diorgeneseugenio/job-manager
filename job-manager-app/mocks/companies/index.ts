import { FETCH_COMPANIES } from "../../api/companies/queries";

export const mockFetchCompanies = {
  request: {
    query: FETCH_COMPANIES,
  },
  result: {
    data: {
      fetchAllCompanies: [
        {
          id: 1,
          name: "Nkey",
          aboutDescription: "Transformando tecnologias.",
          state: "SC",
          city: "Florianópolis",
        },
      ],
    },
  },
};

export const mockCompanies = [
  {
    id: 1,
    name: "Nkey",
    aboutDescription: "Transformando tecnologias.",
    state: "SC",
    city: "Florianópolis",
  },
  {
    id: 2,
    name: "Fluid API",
    aboutDescription: "Transformando integrações.",
    state: "SC",
    city: "Florianópolis",
  },
];
