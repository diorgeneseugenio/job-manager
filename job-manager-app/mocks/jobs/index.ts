import { FETCH_JOBS } from "../../api/jobs/queries";
import { Job } from "../../types/job";

export const mockFetchJobs = {
  request: {
    query: FETCH_JOBS,
  },
  result: {
    data: {
      fetchAllJobs: [
        {
          id: 1,
          companyId: 1,
          title: "Dev",
          description: "Job",
          benefits: "VT;VR",
          skills: "React;Node",
          salary: "2000",
          allowRemote: true,
          ended: false,
          company: {
            name: "Nkey",
            state: "SC",
            city: "Florianópolis",
          },
          createdAt: "",
          updatedAt: "",
        },
      ],
    },
  },
};

export const mockFetchJobsWithError = {
  request: {
    query: FETCH_JOBS,
  },
  error: new Error("Problem to fetch data"),
};

export const mockJob: Job = {
  id: 1,
  companyId: 1,
  title: "Dev",
  description: "Job",
  benefits: "VT;VR",
  skills: "React;Node",
  salary: 2000,
  allowRemote: true,
  ended: false,
  company: {
    id: 1,
    name: "Nkey",
    aboutDescription: "Transformando ideias",
    state: "SC",
    city: "Florianópolis",
  },
  createdAt: "",
  updatedAt: "",
};
