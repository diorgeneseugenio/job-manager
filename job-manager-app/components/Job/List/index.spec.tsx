/**
 * @testEnvironment node
 */
import React from "react";
import { render, waitFor } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";

import JobList from "./";
import { mockFetchJobs, mockFetchJobsWithError } from "../../../mocks/jobs";
import { mockFetchCompanies } from "../../../mocks/companies";

describe("JobList", () => {
  it("renders correctly", async () => {
    const { container, getByRole } = render(
      <MockedProvider
        mocks={[mockFetchJobs, mockFetchCompanies]}
        addTypename={false}
      >
        <JobList />
      </MockedProvider>
    );

    await waitFor(() =>
      expect(getByRole("heading", { name: /Dev/i })).toBeDefined()
    );

    expect(container).toMatchSnapshot();
  });

  it("handles problem to load", async () => {
    const { container, getByText } = render(
      <MockedProvider mocks={[mockFetchJobsWithError]} addTypename={false}>
        <JobList />
      </MockedProvider>
    );

    await waitFor(() =>
      expect(getByText("Não foi possível carregar as vagas")).toBeDefined()
    );

    expect(container).toMatchSnapshot();
  });
});
