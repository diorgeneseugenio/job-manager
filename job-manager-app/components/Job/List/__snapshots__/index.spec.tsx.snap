// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobList handles problem to load 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-rowActions-11 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        <ol
          class="MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Vagas
            </p>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator"
          >
            ›
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
            >
              Listagem
            </p>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root makeStyles-buttonAdd-12 MuiIconButton-colorPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-rowFilter-13 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
            for="companyId"
            id="companyId-label"
          >
            Filtro por empresa
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="companyId-label companyId"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="companyId"
              role="button"
              tabindex="0"
            >
              Selecione uma empresa
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              required=""
              tabindex="-1"
              value="0"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-16 PrivateNotchedOutline-legendNotched-17"
              >
                <span>
                  Filtro por empresa
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-12"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-colorSecondary"
      >
        Não foi possível carregar as vagas
      </p>
    </div>
  </div>
</div>
`;

exports[`JobList renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-rowActions-1 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6"
    >
      <nav
        aria-label="breadcrumb"
        class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        <ol
          class="MuiBreadcrumbs-ol"
        >
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              Vagas
            </p>
          </li>
          <li
            aria-hidden="true"
            class="MuiBreadcrumbs-separator"
          >
            ›
          </li>
          <li
            class="MuiBreadcrumbs-li"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
            >
              Listagem
            </p>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-6"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root makeStyles-buttonAdd-2 MuiIconButton-colorPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-rowFilter-3 MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
            for="companyId"
            id="companyId-label"
          >
            Filtro por empresa
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="companyId-label companyId"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="companyId"
              role="button"
              tabindex="0"
            >
              Selecione uma empresa
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              required=""
              tabindex="-1"
              value="0"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-6 PrivateNotchedOutline-legendNotched-7"
              >
                <span>
                  Filtro por empresa
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-3"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-8 MuiPaper-outlined MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5"
          >
            Dev
             
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeSmall"
              focusable="false"
              title="Aceita remoto"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
              />
            </svg>
          </h5>
          <span
            class="MuiTypography-root makeStyles-company-9 MuiTypography-caption MuiTypography-gutterBottom MuiTypography-displayBlock"
          >
            Nkey
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-gutterBottom"
          >
            Job
          </p>
        </div>
        <div
          class="MuiCardActions-root makeStyles-cardActions-10 MuiCardActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Editar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Deletar
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
